projecttacoma/measure-repository-service-test-kit
# Library data requirements test suite

# Summary
Library $data-requirements test suite

## New behavior
Added test suite tests accepted parameters and expected errors for Library $data-requirements

## Code changes
 - `library_data_requirements.rb` file containing 7 new tests
 - `library_data_requirements_spec.rb` for testing new suite
 - Updated `measure_repository_service_test_kit.rb` to include new test suite

# Testing guidance
 - `rspec`
 - Start up the test kit with the instruction in the readme
 - In `measure-repository-service`
 - `npm run db:reset`
 - To add Library identifier, head into `ecqm-content-r4-2021/bundles/measure/ColorectalCancerScreeningsFHIR/ColorectalCancerScreeningsFHIR-bundle.json` and find the `Library` resource in the bundle with `"id": "ColorectalCancerScreeningsFHIR"`. 
 - Add the following code snippet to the `Library`: 
 ```
        "identifier": {
          "system": "testSystem",
          "value": "testCode"
        }
```
 - `npm run db:loadBundle "ecqm-content-r4-2021/bundles/measure/ColorectalCancerScreeningsFHIR/ColorectalCancerScreeningsFHIR-bundle.json"`
 - `npm start`
 - Navigate to `http://localhost:4567`
 - Select the Library $data-requirements suite
 - Click "run all tests"
 - Enter the following information for each field: 
 - Library id: `ColorectalCancerScreeningsFHIR`
 - FHIR Server Base: `http://localhost:3000`
 - Library url: `http://ecqi.healthit.gov/ecqms/Library/ColorectalCancerScreeningsFHIR`
 - Library Identifier: `testCode`
 - Library version: `0.0.003`
 - Make sure all tests pass!
<img width="541" alt="Screen Shot 2023-02-22 at 7 47 50 AM" src="https://user-images.githubusercontent.com/2643955/220774499-f2e4314d-c936-4c45-b79c-e387cb6a5679.png">
